From 081bd2bf9fd4a103ab65f9e1083b30c3677f0647 Mon Sep 17 00:00:00 2001
From: Ebben Aries <exa@dscp.org>
Date: Mon, 17 Jul 2023 18:44:00 -0700
Subject: [PATCH] fmt9+ compatibility
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Summary:

Pass quic::LocalErrorCode to toString to return folly::StringPiece

/usr/include/fmt/core.h:2817:44:   required from ‘std::string fmt::v10::format(format_string<T ...>, T&& ...) [with T = {const char*, quic::LocalErrorCode, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, const char*, quic::LocalErrorCode, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >]’
/home/exa/repo/github.com/facebookincubator/mvfst/quic/server/QuicServerWorker.cpp:697:27:   required from here
/usr/include/fmt/core.h:1691:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1691 |       formattable,
      |       ^~~~~~~~~~~
---
 quic/server/QuicServerWorker.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/quic/server/QuicServerWorker.cpp b/quic/server/QuicServerWorker.cpp
index 4c32ec26..b414d40e 100644
--- a/quic/server/QuicServerWorker.cpp
+++ b/quic/server/QuicServerWorker.cpp
@@ -696,9 +696,9 @@ PacketDropReason QuicServerWorker::isDstConnIdMisrouted(
     const auto& ex = maybeParsedConnIdParam.error();
     VLOG(3) << fmt::format(
         "Dropping packet due to DCID parsing error={}, errorCode={},"
-        "routingInfo = {} ",
+        "routingInfo={} ",
         ex.what(),
-        ex.errorCode(),
+        toString(ex.errorCode()),
         logRoutingInfo(dstConnId));
     // TODO do we need to reset?
     return PacketDropReason::PARSE_ERROR_DCID;
-- 
2.41.0

